hoomd/md/update.py

```diff
@@ -149,133 +150,140 @@ class mueller_plathe_flow:
 
-        # create the c++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_updater = _md.MuellerPlatheFlow(
-                hoomd.context.current.system_definition, group.cpp_group,
-                flow_target.cpp_variant, slab_direction, flow_direction,
-                n_slabs, min_slab, max_slab)
+    def _attach(self):
+        group = self._simulation.state._get_group(self.filter)
+        sys_def = self._simulation.state._cpp_sys_def
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            self._cpp_obj = _md.MuellerPlatheFlow(
+                sys_def, group, self.flow_target, self.slab_direction,
+                self.flow_direction, self.n_slabs, self.min_slab, self.max_slab,
+                self.flow_epsilon)
         else:
-            self.cpp_updater = _md.MuellerPlatheFlowGPU(
-                hoomd.context.current.system_definition, group.cpp_group,
-                flow_target.cpp_variant, slab_direction, flow_direction,
-                n_slabs, min_slab, max_slab)
-
-        self.setupUpdater(period)
-
-    def get_n_slabs(self):
-        """Get the number of slabs."""
-        return self.cpp_updater.getNSlabs()
-
-    def get_min_slab(self):
-        """Get the slab id of min velocity search."""
-        return self.cpp_updater.getMinSlab()
-
-    def get_max_slab(self):
-        """Get the slab id of max velocity search."""
-        return self.cpp_updater.getMaxSlab()
-
-    def get_flow_epsilon(self):
-        """Get the tolerance between target flow and actual achieved flow."""
-        return self.cpp_updater.getFlowEpsilon()
-
-    def set_flow_epsilon(self, epsilon):
-        """Set the tolerance between target flow and actual achieved flow.
-
-        Args:
-            epsilon (float): New tolerance for the deviation of actual and
-                achieved flow.
-
-        """
-        return self.cpp_updater.setFlowEpsilon(float(epsilon))
+            self._cpp_obj = _md.MuellerPlatheFlowGPU(
+                sys_def, group, self.flow_target, self.slab_direction,
+                self.flow_direction, self.n_slabs, self.min_slab, self.max_slab,
+                self.flow_epsilon)
+        super()._attach()
 
-    def get_summed_exchanged_momentum(self):
-        """Returned the summed up exchanged velocity of the full simulation."""
-        return self.cpp_updater.getSummedExchangedMomentum()
+    @property
+    def summed_exchanged_momentum(self):
+        R"""Returned the summed up exchanged velocity of the full simulation."""
+        if self._attached:
+            return self._cpp_obj.summed_exchanged_momentum
+        else:
+            return None
 
+    @property
     def has_min_slab(self):
-        """Returns, whether this MPI instance is part of the min slab."""
-        return self.cpp_updater.hasMinSlab()
+        R"""Returns, whether this MPI instance is part of the min slab."""
```

Comment:
Remove `has_min_slab` and `has_max_slab`. I don't see an obvious reason why the user would need to query this. If it gets requested later, this is easy enough to add back in.